File: alg/busca_binaria.py
def busca_binaria(vetor, elemento):
    inicio = 0
    fim = len(vetor) - 1

    while inicio <= fim:
        meio = (inicio + fim) // 2

        if vetor[meio] == elemento:
            return meio
        elif vetor[meio] > elemento:
            fim = meio - 1
        else:
            inicio = meio + 1

    return "O elemento não está presente na lista"

vetor = [i for i in range(3000)]
    
elemento = int(input())

print(busca_binaria(vetor, elemento))
# 1 (índice do elemento)

File: C/exercicios/buscas/busca_binaria.c
#include<stdio.h>

void busca_binaria(int array[], int tam, int elemento){
    int inicio = 0, meio = 0, fim = tam;
    for (int i = 0; i < tam; i++){
        meio = (inicio+fim)/2;

        if(array[meio] == elemento){
            printf("Você achou o seu elemento, ele está na posição: %d\n\n", meio);
            return;
        }else if(array[meio] > elemento){
            fim = meio - 1;
        }else{
            inicio = meio + 1;
        }
    }
}


int main(){
    int vetor[4] = {1, 3, 4, 5};


    busca_binaria(vetor, 4, 3);
    return 0;
}